File: python/소수 만들기.py
from itertools import combinations

def solution(nums):
    def is_prime(x):
        if x < 2:
            return False
        for i in range(2, int(x**0.5)+1):
            if x % i == 0:
                return False
        return True
    
    cnt = 0
    for j in combinations(nums, 3):
        if is_prime(sum(j)):
            cnt += 1
    return cnt

File: js/소수 만들기.js
function isPrime(n) {
    if (n < 2) {
        return false
    }
    for (let i=2; i <= parseInt(n**0.5); i++) {
        if (n % i === 0) {
            return false
        }
    }
    return true
}

function solution(nums) {
    arr = [];
    let cnt = 0;
    for (let i=0; i<nums.length; i++) {
        for (let j=i+1; j<nums.length; j++) {
            for (let k=j+1; k<nums.length; k++) {
                arr.push([nums[i], nums[j], nums[k]]);
            }
        }
    }
    for (let i=0; i<arr.length; i++) {
        let tmp = arr[i].reduce((acc,cur) => acc+cur, 0);
        if (isPrime(tmp)) {
            cnt += 1;
        }
    }
    return cnt;
}